airbyte-integrations/connectors/source-sftp/src/main/java/io/airbyte/integrations/source/sftp/SftpSource.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.sftp;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.util.AutoCloseableIterator;
+import io.airbyte.commons.util.AutoCloseableIterators;
+import io.airbyte.integrations.BaseConnector;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.base.Source;
+import io.airbyte.protocol.models.AirbyteCatalog;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.SyncMode;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SftpSource extends BaseConnector implements Source {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SftpSource.class);
+
+    public static void main(String[] args) throws Exception {
+        new IntegrationRunner(new SftpSource()).run(args);
+    }
+
+    /**
+     * Check SFTP connection status and existence of working directory if set
+     * with the provided Json configuration.
+     *
+     * @param config - json configuration for connecting SFTP
+     * @return AirbyteConnectionStatus status of the connection result.
+     * @throws Exception - any exception.
+     */
+    @Override
+    public AirbyteConnectionStatus check(JsonNode config) throws Exception {
```

Comment:
From what I can see, this method actually can never throw `Exception`, because all code that can throw an exception is inside `try-catch` block.

Reply:
Thanks for noticing. I'll remove it